source/helpers.coffee

```diff
@@ -6,12 +6,19 @@ defineTypeWithFields = (BaseType, namespace, className, fields) ->
   SubType = BaseType.extend(classPath)
   SubType::fields = -> _.extend BaseType::fields(), fields
   Space.resolvePath(namespace)[className] = SubType
+  definedSerializable = {}
+  definedSerializable[classPath] = SubType
+  return definedSerializable
+
 
 Space.messaging.define = (BaseType, options..., definitions) ->
 
   if !BaseType.isSerializable then throw new Error 'BaseType must extend Space.messaging.Serializable'
   if !definitions? then throw new Error "Space.messaging.define is missing definitions for #{BaseType}."
   if options.length is 1 then namespace = options[0].toString() else namespace = ''
-  defineTypeWithFields(BaseType, namespace, className, fields) for className, fields of definitions
+  definedSerializables = {}
+  _.extend definedSerializables, defineTypeWithFields(BaseType, namespace, className, fields) for className, fields of definitions
```

Comment:
It may also make sense to replace the loop with an array function. This is what we would do if moving away from CS